CustomEventEmitter.js

```diff
@@ -0,0 +1,100 @@
+class EventEmitter {
+  listeners = {}; // key-value pair
+
+  addListener(eventName, fn) {
+    if (this.listeners[eventName]) {
+      this.listeners[eventName].add(fn);
+    } else {
+      this.listeners[eventName] = new Set();
+      this.listeners[eventName].add(fn);
+    }
+  }
+
+  on(eventName, fn) {
+    this.addListener(eventName, fn);
+  }
```

Comment:
Actually in official EventEmitter the `addListener` is an alias for `on` like mentioned [here](https://d17btkcdsmqrmh.cloudfront.net/node-gmp/docs/event-emitter-vs-buffers-vs-streams/event_emitter#addlistener) . So It is not an issue, but just an information. Based on that knowledge, the logic inside `addListener` should be in `on`.